hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/ConfigProviderImpl.java

```diff
@@ -272,16 +193,93 @@ private ConfigurationBuilder addConfigData(@NonNull final ConfigurationBuilder b
                 .withConfigDataType(TraceabilityConfig.class)
                 .withConfigDataType(UpgradeConfig.class)
                 .withConfigDataType(UtilPrngConfig.class)
-                .withConfigDataType(VirtualdatasourceConfig.class);
+                .withConfigDataType(VersionConfig.class)
+                .withConfigDataType(VirtualdatasourceConfig.class)
+                .withConverter(new CongestionMultipliersConverter())
+                .withConverter(new EntityScaleFactorsConverter())
+                .withConverter(new EntityTypeConverter())
+                .withConverter(new KnownBlockValuesConverter())
+                .withConverter(new LegacyContractIdActivationsConverter())
+                .withConverter(new MapAccessTypeConverter())
+                .withConverter(new RecomputeTypeConverter())
+                .withConverter(new ScaleFactorConverter())
+                .withConverter(new AccountIDConverter())
+                .withConverter(new ContractIDConverter())
+                .withConverter(new FileIDConverter())
+                .withConverter(new HederaFunctionalityConverter())
+                .withConverter(new ProfileConverter())
+                .withConverter(new SidecarTypeConverter())
+                .withConverter(new SemanticVersionConverter())
+                .withConverter(new KeyValuePairConverter())
+                .withConverter(new BytesConverter())
+                .withValidator(new EmulatesMapValidator());
     }
 
-    @NonNull
-    private ConfigurationBuilder addValidators(@NonNull final ConfigurationBuilder builder) {
-        return builder.withValidator(new EmulatesMapValidator());
+    private void addByteSource(@NonNull final ConfigurationBuilder builder, @NonNull final Bytes propertyFileContent) {
+        requireNonNull(builder);
+        requireNonNull(propertyFileContent);
+        try (final var in = propertyFileContent.toInputStream()) {
+            final byte[] bytes = in.readAllBytes();
+            try (ByteArrayInputStream inputStream = new ByteArrayInputStream(bytes)) {
+                Properties properties = new Properties();
+                properties.load(inputStream);
+                final PropertyConfigSource configSource = new PropertyConfigSource(properties, 101);
+                builder.withSource(configSource);
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException("Can not create config source for bytes", e);
+        }
     }
 
-    @Override
-    public VersionedConfiguration getConfiguration() {
-        return configuration;
+    private void addFileSources(@NonNull final ConfigurationBuilder builder, final boolean useGenesisSource) {
```

Comment:
Can we really ignore the use case with a unconfigured `useGenesisSource` boolean? I assume that Dagger would pass `null` in that case and that would end in an outboxing error.

Reply:
Agreed. The bigger change from which this PR came will remove the @Inject and use little-b boolean, but for this PR, it should be ready for a null. Even if only for 1 day :)